direct/src/directnotify/DirectNotify.py

```diff
@@ -77,40 +83,42 @@ def setDconfigLevel(self, categoryName):
             level = 'error'
 
         category = self.getCategory(categoryName)
+        assert category is not None
```

Comment:
I see no guarantees here that `category` will not be `None`. I believe it would be better to raise an Error instead. The previous code would have already failed with an attribute error of some kind (None type object has no attribute...). Neither that now an assertion error are particularly friendly. Raising an error and telling the user that the supplied category name did not exist is much more informative, and it will still provide the necessary type-narrowing that this assert is doing.

Reply:
Yeah, that makes sense. What kind of error would you suggest?

Reply:
Okay, we'll leave it out for now then.

Reply:
Even just adding an explanation to the assert would probably do: `assert category is not None, f'failed to find category: {categoryName}'`